src/Chainweb/Pact/TransactionExec.hs

```diff
@@ -429,13 +431,11 @@ applyUpgrades v cid height
 
       infoLog $ "Running upgrade tx " <> sshow (_cmdHash tx)
 
-      r <- try $ runGenesis tx permissiveNamespacePolicy interp
-
-      case r of
+      tryAllSynchronous (runGenesis tx permissiveNamespacePolicy interp) >>= \case
         Right _ -> return ()
-        Left (e :: SomeException) -> do
+        Left e -> do
           logError $ "Upgrade transaction failed! " <> sshow e
-          return ()
+          void $ throwM e
```

Comment:
I consider this line the only actual change to production of this PR (beside the update of the pact pin).